content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,501 @@
+
+```
+PHP Warning:  Unknown: [ddappsec] Connection to helper failed and we are not going to attempt to launch it: dd_error
+```
+
+The warning could be followed by one of these error messages:
+
+```
+PHP Warning:  Unknown: [ddappsec] Could not open lock file /tmp/ddappsec.lock: Permission denied in Unknown on line 0
+```
+```
+PHP Warning:  Unknown: [ddappsec] Call to bind() failed: Permission denied
+```
+```
+PHP Warning:  Unknown: [ddappsec] Failed to unlink /tmp/ddappsec.sock: Operation not permitted
+```
+
+This indicates that the lock file or socket file used by the extension has invalid permissions, or the user executing the PHP process does not have write access to the `tmp` directory. 
+
+If the lock file or socket file has invalid permissions, you can either delete them and restart Apache/FPM or adjust the `user:group` to match the one used by Apache/FPM, for example, `www-data`.
+
+If the user doesn’t have write access to the tmp directory, you can change the location of the lock file and socket file by modifying the following settings in the extension’s `.ini` file:
+
+```
+ddappsec.helper_socket_path = /<directory with compatible permissions>/ddappsec.sock
+ddappsec.helper_lock_path = /<directory with compatible permissions>/ddappsec.lock
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+#### Confirm AppSec is enabled in the running application
+
+[Tracer startup logs][1] show the tracer configuration and whether Application Security is enabled or not. If `appsec` is `true`, then Application Security is enabled and running.
+
+For example, the following startup log shows that Application Security is disabled:
+
+```
+2022/02/17 14:49:00 Datadog Tracer v1.36.0 INFO: DATADOG TRACER CONFIGURATION {"date":"2022-02-17T14:49:00+01:00","os_name":"Linux (Unknown Distribution)","os_version":"5.13.0","version":"v1.36.0","lang":"Go","lang_version":"go1.17.1","env":"prod","service":"grpcserver","agent_url":"http://localhost:8126/v0.4/traces","debug":false,"analytics_enabled":false,"sample_rate":"NaN","sampling_rules":null,"sampling_rules_error":"","service_mappings":null,"tags":{"runtime-id":"69d99219-b68f-4718-9419-fa173a79351e"},"runtime_metrics_enabled":false,"health_metrics_enabled":false,"profiler_code_hotspots_enabled":false,"profiler_endpoints_enabled":false,"dd_version":"","architecture":"amd64","global_service":"","lambda_mode":"false","appsec":false,"agent_features":{"DropP0s":false,"Stats":false,"StatsdPort":0}}
+```
+
+#### Enable debug logs
+
+Enable debug logs with the environment variable `DD_TRACE_DEBUG=1`. The Application Security library will log to the standard error output.
+
+**Note:** Application Security only outputs logs when it is enabled. Use the environment variable `DD_APPSEC_ENABLED=1` to enable Application Security.
+
+
+[1]: /tracing/troubleshooting/tracer_startup_logs/
+{{< /programming-lang >}}
+{{< programming-lang lang="NodeJS" >}}
+
```

Comment:
Somewhere around here maybe add a yellow warning box saying something like
`If you've upgraded your library from 1.x to 2.x, you may need to assess breaking changes with this guide: https://github.com/DataDog/dd-trace-js/blob/master/MIGRATING.md`
You can change the wording if you want, and put it in another page if you think it's more relevant.

Reply:
Great, I approved your other PR too!